bin/pre-push

```diff
@@ -1,16 +1,106 @@
 #!/bin/sh
 
+remote="$1"
+url="$2"
+sha40=0000000000000000000000000000000000000000
+
 echo "\nBy contributing to this project, you license the materials you contribute under the GNU General Public License v2 (or later). All materials must have GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n"
 
-current_branch=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
-if [ 'master' = $current_branch ]
-then
-    read -p "You're about to push ¡¡¡[ master ]!!!, is that what you intended? [y|n] " -r < /dev/tty
-    echo
-    if echo $REPLY | grep -E '^[Yy]$' > /dev/null
-    then
-        exit 0 # push will execute
-    fi
-    exit 1 # push will not execute
-fi
+# We want to know whether this script was called from the CLI or from a GUI app.
+# If it is the later, we would only show the ESLint command to get the report,
+# not the report itself, because GUIs tend to left a small space for pre-commit
+# messages and they are not useful. At least this way, the user can see something.
+#
+# - First, get the caller: the process that executes the commit.
+#	The command tree would be: caller > git > pre-commit script
+#       so we need to get the grandparent of this script
+#
+# - Second, if the caller is not one of the available shells (see /etc/shells),
+#	we are going to assume that it is called from a GUI app.
+
+# caller=$(which $(ps -o comm=  $(ps -o ppid= $(ps -o ppid= $$))))
+caller=$(which $(ps -o comm=  $(ps -o ppid= $PPID)))
```

Comment:
oh, I thought this was valid POSIX :(

For now, 905e8ec removes the `is_cli` logic. It could be made simpler and more robust with the suggestions by @ehg in later iterations within other PRs.